src/frontend/efiling-frontend/src/components/page/home/Home.js

```diff
@@ -65,32 +61,33 @@ const checkCSOAccountStatus = (
 ) => {
   axios
     .get(`/submission/${submissionId}`)
-    .then(({ data: { userDetails, navigation, clientApplication } }) => {
+    .then(({ data: { navigation, clientApplication } }) => {
       setClientApplicationName(clientApplication.displayName);
-      saveDataToSessionStorage(userDetails.internalClientNumber, navigation);
-
-      if (userDetails.accounts) {
-        const csoAccountIdentifier = userDetails.accounts.find(
-          (o) => o.type === "CSO"
-        ).identifier;
-        sessionStorage.setItem("csoAccountId", csoAccountIdentifier);
-        setCsoAccountStatus({ isNew: false, exists: true });
-      } else {
-        axios
-          .get("/bceidAccount")
-          .then(({ data: { firstName, middleName, lastName } }) => {
-            setRequiredState(
-              firstName,
-              middleName,
-              lastName,
-              setApplicantInfo,
-              setShowLoader
+      saveNavigationToSessionStorage(navigation);
+
+      axios
+        .get("/csoAccount")
+        .then(({ data: { clientId, internalClientNumber } }) => {
+          sessionStorage.setItem("internalClientNumber", internalClientNumber);
+          sessionStorage.setItem("csoAccountId", clientId);
+          setCsoAccountStatus({ isNew: false, exists: true });
+        })
+        .catch(() => {
```

Comment:
is it on any http status ? should only be on 404 right ?

Reply:
this is on any error status, will make check for specific case